Frontend/CO.CDP.OrganisationApp/Pages/Registration/YourDetails.cshtml.cs

```diff
@@ -76,23 +77,37 @@ public async Task<IActionResult> OnPost()
                 lastName: LastName
             ));
         }
-        catch (ApiException aex)
-            when (aex is ApiException<Person.WebApiClient.ProblemDetails> pd)
+        catch (ApiException<Person.WebApiClient.ProblemDetails> aex)
         {
-            Error = pd.Result.Detail;
+            var statusCode = aex.StatusCode;
+            var problemDetails = aex.Result;
+
+            switch (statusCode)
+            {
+                case StatusCodes.Status400BadRequest:
+                    ModelState.AddModelError(string.Empty, ErrorMessagesList.DuplicatePersonName);
```

Comment:
What if we add more validation rules to the Person API? 400 will no longer map to a single error. Wouldn't it be better to inspect the actual error type in `ProblemDeatils` and map it to the error message?

Reply:
Than potentially code can be written as 

```
catch (ApiException<Person.WebApiClient.ProblemDetails> aex)
    when (aex.StatusCode == StatusCodes.Status400BadRequest)
{
    var message = aex.Result.Detail switch
    {
        "Err1" => "ui_error_message",
        _ => "default_ui_error_message",
    };

    ModelState.AddModelError(string.Empty, message);
}
```

Reply:
@JBaigGoaco this comment is resolved but there is still work to be done to map error from API.
Are you creating another story for this?

Reply:
yes sure. I will add another story on in the sprint backlog with our discussion and comments. to cover the error mapping from the API side of errors. For example for 400 we need to map error codes returned from api, There could be different messages for 400 error.

Reply:
Story created on the Borad: https://noticingsystem.atlassian.net/browse/DP-168